### PR TITLE
add shear production to diagnostic edmf

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Adds shear production to diagnostic edmf. Only includes vertical gradients for now.
#1897 

## To-do

- [ ] fix allocation
- [x] Add rico

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/cache/diagnostic_edmf_precomputed_quantities.jl [hunk 1]
```diff
@@ -484,9 +484,14 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
             Y.c.ρ,                                                             # ρ
         ),
     )
-    @. ᶜshear² = FT(1e-4)
+
+    ᶠu⁰ = p.ᶠtemp_C123
+    @. ᶠu⁰ = C123(ᶠinterp(Y.c.uₕ)) + C123(ᶠu³⁰)
+    ct3_unit = p.ᶜtemp_CT3
+    @. ct3_unit = CT3(Geometry.WVector(FT(1)), ᶜlg)
+    @. ᶜshear² = norm_sqr(adjoint(CA.ᶜgradᵥ(ᶠu⁰)) * ct3_unit)
```
Comment (szy21) [inline src/cache/diagnostic_edmf_precomputed_quantities.jl:492]: @charleskawczynski I copied this from the old EDMF [here](https://github.com/CliMA/ClimaAtmos.jl/blob/199f4c90617bb5fb90148a946e31ce9134030627/src/TurbulenceConvection_deprecated/update_aux.jl#L415-L417) but it allocates a lot. Is there a way to not allocate here?

Comment (charleskawczynski) [reply src/cache/diagnostic_edmf_precomputed_quantities.jl:492]: There's not really anything wrong with this code-- these allocations are due to ClimaCore issues, some hopefully will be fixed soon

Comment (szy21) [reply src/cache/diagnostic_edmf_precomputed_quantities.jl:492]: Ah, ok. It increases the allocation by a factor of 55 though:( Does this look like the simplest way to calculate the approximated shear production (∂u∂z^2 + ∂v∂z^2 + ∂w∂z^2)?

**Pull request merged**